### ISSUE TITLE
Change in behavior in inline variable handling between clang version

### ISSUE THREAD

**Issue by jfalcou** [llvm/llvm-project#86520]
Repro code : 
https://godbolt.org/z/bnbof95oo

Removing inline from the declaration of my_wrapper seems to cause the issue to go away.
ASAN and UB santiizers report nothing

**Comment by vitalybuka** [comment]
There is UB report https://godbolt.org/z/3n8zs5vqz

**Comment by vitalybuka** [comment]
> There is UB report https://godbolt.org/z/3n8zs5vqz

But it looks bogus.

**Comment by jfalcou** [comment]
Yeah the fact the pointer is null is the actual issue I think. It wasn't there in older clang version.

**Comment by shafik** [comment]
Looks like this stated with clang-9: https://godbolt.org/z/8Y4eaEYcj

CC @AaronBallman
